evm/src/executor/fork/init.rs

```diff
@@ -42,6 +43,11 @@ where
         block
     } else {
         if let Ok(latest_block) = provider.get_block_number().await {
+            // If the user is forking from an older block in a non-archive node, the `get_block`
+            // call will return null.
+            if block_number < latest_block.as_u64() {
```

Comment:
ditto on the 128/64 block retrieval threshold—i think forking from blocks less than these distances is ok.

Reply:
Resolving per https://github.com/foundry-rs/foundry/pull/5131#discussion_r1234240848